### PR TITLE
Added Grid coordinates variant of RS_Value

### PR DESCRIPTION
**Pull request by prantogg**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- No

## What changes were proposed in this PR?

Added a variant of RS_Value to accept grid coordinates of raster

## How was this patch tested?

Passing existing tests

## Did this PR include necessary documentation updates?

- Yes, I have updated the documentation update.

### CODE REVIEW

File: common/src/main/java/org/apache/sedona/common/raster/PixelFunctions.java [hunk 1]
```diff
@@ -41,6 +42,32 @@ public static Double value(GridCoverage2D rasterGeom, Geometry geometry, int ban
         return values(rasterGeom, Collections.singletonList(geometry), band).get(0);
     }
 
+    public static Double value(GridCoverage2D rasterGeom, int colX, int rowY, int band) throws TransformException {
+        int numBands = rasterGeom.getNumSampleDimensions();
+        if (band < 1 || band > numBands) {
+            // Invalid band index. Return nulls.
+            return null;
+        }
```
Comment (furqaankhan) [inline common/src/main/java/org/apache/sedona/common/raster/PixelFunctions.java:50]: Please use the RasterUtil.ensureBand method for this

Comment (prantogg) [reply common/src/main/java/org/apache/sedona/common/raster/PixelFunctions.java:50]: Using RasterUtil.ensureBand will throw an IllegalArgumentException. For RS_Value we expect to return null for invalid bands.

Comment (furqaankhan) [reply common/src/main/java/org/apache/sedona/common/raster/PixelFunctions.java:50]: Is there a reason for it?

File: common/src/main/java/org/apache/sedona/common/raster/PixelFunctions.java [hunk 2]
```diff
@@ -41,6 +42,32 @@ public static Double value(GridCoverage2D rasterGeom, Geometry geometry, int ban
         return values(rasterGeom, Collections.singletonList(geometry), band).get(0);
     }
 
+    public static Double value(GridCoverage2D rasterGeom, int colX, int rowY, int band) throws TransformException {
```
Comment (furqaankhan) [inline common/src/main/java/org/apache/sedona/common/raster/PixelFunctions.java:45]: @jiayuasu I think we should have `excludeNoData` flag.

**Pull request closed without merging**